sdk/eventhub/Azure.Messaging.EventHubs/src/Core/TimeSpanExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Messaging.EventHubs.Core
+{
+    /// <summary>
+    ///   The set of extensions for the <see cref="TimeSpan" />
+    ///   class.
+    /// </summary>
+    ///
+    internal static class TimeSpanExtensions
+    {
+        /// <summary>
+        ///   Calculates the duration remaining in a given period, with consideration to
+        ///   the amount of time that has already elapsed.
+        /// </summary>
+        ///
+        /// <param name="instance">The instance that this method was invoked on.</param>
+        /// <param name="elapsed">The amount of time that has already elapsed.</param>
+        ///
+        /// <returns>The amount of time remaining in the time period.</returns>
+        ///
+        public static TimeSpan CalculateRemaining(this TimeSpan instance,
```

Comment:
Pretty convenient. I may use it in `Event Processor` if your PR gets merged before mine.